src/Views/scstormrunoff.html

```diff
@@ -134,10 +134,131 @@ <h4>Citation</h4>
+                                            </div>
+                                            <div class="flow-form-button">
+                                                <button ng-click="vm.openAddFlowSegment($index)"><i class="fa fa-arrow-left"></i> Add Flow Segment</button>
+                                            </div>
+                                        </div>
+                                        <div ng-if="vm.chosenFlowTypeIndex == $index">
+                                            <h4 class="modal-title">New {{flowType.displayName}} Segment</h4>
+                                            <ng-form name="{{flowType.id}}" class="hidden-print">
+                                                <div ng-repeat="question in flowType.questions">
+                                                    <label for="{{question.label}}">{{question.label}}</label>
+                                                    <span ng-show="{{flowType.id}}.{{question.id}}.$error.pattern" class="red label">
+                                                        <span class="fa fa-exclamation-triangle"></span>&nbsp;{{question.invalidMessage}}
+                                                    </span>
+                                                    <input ng-if="question.type != 'select' && question.pattern == 'greaterThanZero'" type="{{ question.type }}" name="{{question.id}}" ng-model="question.value" ng-pattern="greaterThanZero" required/>
+                                                    <input ng-if="question.type != 'select' && question.pattern == 'greaterThanOrEqualToZero'" type="{{ question.type }}" name="{{question.id}}" ng-model="question.value" ng-pattern="greaterThanOrEqualToZero" required/>
+                                                    <input ng-if="question.type != 'select' && question.pattern == null" type="{{ question.type }}" name="{{question.id}}" ng-model="question.value" required/>
+                                                    <select ng-if="question.type == 'select'" name="{{question.id}}" ng-model="question.value" ng-options="key for (key, value) in question.options track by value"></select>                                                                              
+                                                </div>
+                                                <div>&nbsp;</div>
+                                                <div class="form-row-extremes">
+                                                    <div class="form-column-left">
+                                                        <button ng-click="vm.closeAddFlowSegment()"><i class="fa fa-times"></i> Cancel</button>
+                                                    </div>
+                                                    <div class="form-column-right">
+                                                        <button ng-click="vm.addFlowSegment()"><i class="fa fa-plus"></i> Add</button>
```

Comment:
I'm able to click "Add" and successfully add a flow segment, even if I haven't completed all the form elements for that flow segment. We should prevent the user from adding flow segments when not all the inputs are provided.